### DISCUSSION TITLE
Using difference between background and image to "segment" the object.

### DISCUSSION THREAD

**Discussion by VivitionDeveloper** [libvips/libvips#4253] [Q&A] [answered]
I'm looking at a simple means of segmenting a car out of an image. I've got an image containing only the background, and the image with the car in it. My naive approach would be to grab the difference of the two images, so I did that, with the following result:
![image](https://github.com/user-attachments/assets/3336c891-5dc6-45f1-a3eb-0797b324a718)

I did this with the following line in python:
diff = np.abs(np.array(image_empty) - image)
where image_empty is the image of the empty photo studio, and image is the one with the car.

As you can see, the blue car works perfectly fine, and with some floodfill or other techniques we could get this to work. The white car however, that's a different story.

Also, I'm getting lines where the seams in the wall are.

What would be a way of making this work properly?

**Comment by VivitionDeveloper** [comment]
Some images I used for testing:
![000_2039a0f7-e32a-4698-9e8b-3b297b181fb4](https://github.com/user-attachments/assets/fe739b98-9676-4dd8-abcb-33af1b28d59a)
![080_175f05de-5c21-435e-8470-e3320b56271f](https://github.com/user-attachments/assets/09ad3142-3266-4134-ac00-703dc7bf0349)
![040_01f1701f-a4d7-4bc8-87d1-f78f8785e01d](https://github.com/user-attachments/assets/f6783397-f59e-4832-a5b5-68b2ced1e771)
![000_cca79b0c-6236-41f8-8686-542e8d5204fe](https://github.com/user-attachments/assets/d14dbcbd-296f-4365-a199-512739543d7b)
![040_d74c9352-5f10-4749-bfc4-e45e2411a931](https://github.com/user-attachments/assets/9eb38bfc-ec5b-428e-9099-4429c43c3fe5)
![080_cbde0e4a-943f-4bf1-aba5-273ddc03d5d9](https://github.com/user-attachments/assets/be7e0153-9442-42a7-8967-3a9bf9408d41)
![000_ca4dfef3-8ca9-4af5-b0b0-2018c1652e6b](https://github.com/user-attachments/assets/8423abb9-03ab-4fca-8e00-a0eef8317434)
![040_998cb66f-470b-4f19-bf80-1ab22acb3d09](https://github.com/user-attachments/assets/5a0ffc80-ca0c-4d06-906b-3cf7484617d2)
![080_1012a67d-770b-4631-83da-e3f9023c8d20](https://github.com/user-attachments/assets/63fc2284-6172-4c8d-ae6b-80f7416d1578)

**Comment by jcupitt** [comment, marked as answer]
I tried quickly with the van, but a white van against a white background is tricky to separate.

Photographers generally use greenscreen for this kind of thing, I expect you know:

https://www.theschoolofphotography.com/tutorials/how-to-use-a-green-screen

Since it's a horrible colour no one is going to want to keep. Perhaps you could drape the back walls with green cloth?

**Reply by VivitionDeveloper** [reply]
I'll take that as: not possible. That's what I figured. Thank you for your honest answer!
